tools/@aws-cdk/spec2cdk/test/services.test.ts

```diff
@@ -25,3 +25,147 @@ test('can codegen service with arbitrary suffix', () => {
   expect(rendered).toContain('function convertCfnApplicationV2PropsToCloudFormation');
   expect(rendered).toContain('function CfnApplicationV2ApplicationCodeConfigurationPropertyValidator');
 });
+
+test('resource interface when primaryIdentifier is an attribute', () => {
+  const service = db.lookup('service', 'name', 'equals', 'aws-voiceid').only();
+
+  const ast = AstBuilder.forService(service, { db });
+
+  const rendered = renderer.render(ast.module);
+
+  expect(rendered).toContain([
+    '/**',
+    ' * Shared attributes for both `CfnDomain` and `Domain`.',
```

Comment:
This implies knowledge of L2s, which doesn't make sense yet. Don't refer to it as "shared attributes" I guess